types/google-adwords-scripts/index.d.ts

```diff
@@ -0,0 +1,1564 @@
+declare namespace AdWordsApp {
+    function adCustomizerSources(): AdWordsSelector<AdCustomizerSource>;
+    function adGroups(): AdWordsSelector<AdGroup>;
+    function adGroupTargeting(): AdWordsTargeting<SearchAdGroupAudience, SearchAdGroupExcludedAudience>;
+    function adMedia(): AdMedia;
+    function adParams(): AdWordsSelector<AdParam>;
+    function ads(): AdWordsSelector<Ad>;
+    function biddingStrategies(): AdWordsSelector<BiddingStrategy>;
+    function budgetOrders(): AdWordsSelector<BudgetOrder>;
+    function budgets(): AdWordsSelector<Budget>;
+    function bulkUploads(): BulkUploads;
+    function campaigns(): AdWordsSelector<Campaign>;
+    function createLabel(name: string, description?: string, backgroundColor?: string): void;
+    function currentAccount(): CurrentAccount;
+    function display(): Display;
+    function excludedPlacementLists(): AdWordsSelector<ExcludedPlacementList>;
+    function extensions(): AdWordsAdExtensions;
+    function getExecutionInfo(): ExecutionInfo;
+    function keywords(): AdWordsSelector<Keyword>;
+    function labels(): AdWordsSelector<Label>;
+    function negativeKeywordLists(): AdWordsSelector<NegativeKeywordList>;
+    function newAdCustomizerSourceBuilder(): AdCustomizerSourceBuilder<AdCustomizerSource>;
+    function newExcludedPlacementListBuilder(): ExcludedPlacementListBuilder<ExcludedPlacementList>;
+    function newNegativeKeywordListBuilder(): NegativeKeywordListBuilder<NegativeKeywordList>;
+    function productAds(): AdWordsSelector<AdWordsEntity>; // TODO: ProductAd
+    function productGroups(): AdWordsSelector<AdWordsEntity>; // TODO: ProductGroup
+    function report(query: string, options?: ReportOptionArguments): AdWordsReport;
+    function shoppingAdGroupTargeting(): any; // TODO: AccountShoppingAdGroupTargeting
+    function shoppingAdGroups(): AdWordsSelector<AdWordsEntity>; // TODO: ShoppingAdGroup
+    function shoppingCampaignTargeting(): any; // TODO: AccountShoppingCampaignTargeting
+    function shoppingCampaigns(): AdWordsSelector<AdWordsEntity>; // TODO: ShoppingCampaign
+    function targeting(): AdWordsSelector<Targeting>;
+    function userlists(): AdWordsSelector<UserList>;
+    function videoAdGroups(): AdWordsSelector<AdWordsEntity>; // TODO: VideoAdGroup
+    function videoAds(): AdWordsSelector<AdWordsEntity>; // TODO: VideoAd
+    function videoCampaigns(): AdWordsSelector<AdWordsEntity>; // TODO: VideoCampaign
+    function videoTargeting(): any; // TODO: VideoTargeting
+}
+
+declare namespace MccApp {
+    function accountLabels(): AdWordsSelector<AdWordsEntity>; // TODO: AccountLabel
+    function accounts(): AdWordsSelector<AdWordsEntity>; // TODO: ManagedAccount
+    function createAccountLabel(name: string): void;
+    function select(account: AdWordsEntity): void; // TODO: ManagedAccount
+}
+
+declare namespace Analytics { }
+
+declare namespace BigQuery { }
+
+declare namespace Calendar { }
+
+declare namespace FusionTables { }
```

Comment:
They are the Advanced APIs available. I wanted to make sure using them didn't throw an error at least until they can be fleshed out with their methods. More info about the Advanced APIs is here: https://developers.google.com/adwords/scripts/docs/features/advanced-apis